py/desispec/scripts/pipe_prod.py

```diff
@@ -20,6 +20,43 @@
 from .. import pipeline as pipe
 
 
+def job_size(ntask, taskprocs, tasktime, nodeproc, maxnodes):
+    """
+    Internal helper function used only in the desi_pipe
+    script to compute the size of batch jobs.
+    """
+    maxprocs = nodeproc * maxnodes
+
+    fullprocs = ntask * taskprocs
+
+    nodes = None
+    time = None
+    procs = None
+    
+    if fullprocs <= maxprocs:
+        # we can run fully unpacked, with every process group
+        # assigned one task.
+        time = tasktime
+        procs = fullprocs
+    else:
+        # we want to repack as efficiently as possible
+        nworker = maxprocs // taskprocs
+
+        taskperworker = ntask // nworker
+        if ntask % nworker != 0:
+            taskperworker += 1
+
+        nworker = ntask // taskperworker
+        if ntask % nworker != 0:
+            nworker += 1
+
+        time = tasktime * taskperworker
+        procs = nworker * taskprocs
+    
+    nodes = procs // nodeproc + 1
```

Comment:
If `procs % nodeproc == 0` then you will end up requesting one more node than you actually need, correct?
Should this be `nodes = max(1, (procs-1) // nodeproc)` instead?

Having writing similar code in the past and been humbled by how many different ways I screwed up the edge cases, I would recommend writing some unit tests to check cases, especially cases like number of processes exactly fitting on an integer number of nodes, just barely exceeding or just barely under subscribing, and just barely exceeding maxnodes.  This looks close enough to right that we could proceed with a merge, but eventually we'll want to check this more closely.